shared/js/background/chrome-events.es6.js

```diff
@@ -271,6 +274,54 @@ chrome.webRequest.onBeforeSendHeaders.addListener(
     ['blocking', 'requestHeaders']
 )
 
+/*
+ * Truncate the referrer header according to the following rules:
+ *   Don't modify the header when:
+ *   - If the header is blank, it will not be modified.
+ *   - If the referrer domain OR request domain are safe listed, the header will not be modified
+ *   - If the referrer domain and request domain are part of the same entity (as defined in our
+ *     entities file for first party sets), the header will not be modified.
+ *
+ *   Modify the header when:
+ *   - If the destination is in our tracker list, we will trim it to eTLD+1 (remove path and subdomain information)
+ *   - In all other cases (the general case), the header will be modified to only the referrer origin (includes subdomain).
+ */
+let ReferrerListenerOptions = ['blocking', 'requestHeaders']
+if (browser !== 'moz') {
+    ReferrerListenerOptions.push('extraHeaders') // Required in chrome type browsers to receive referrer information
+}
+
+chrome.webRequest.onBeforeSendHeaders.addListener(
+    function limitReferrerData (e) {
+        let referrer = e.requestHeaders.filter(header => header.name.toLowerCase() === 'referer')[0] || ''
+        if (referrer) {
+            referrer = referrer.value
+        } else {
+            return
+        }
+
+        // Check if origin is safe listed
+        const tab = tabManager.get({ tabId: e.tabId })
```

Comment:
we are already doing all the checks inside of `truncateReferrer` so I wonder if we really need this one?

Reply:
I think the only check that isn't done in the truncateReferrer is against the broken site list. It could probably be done in truncateReferrer as well if we'd rather put everything there, but it would need to be refactored to accept a tab parameter.

Reply:
Hm, yeah, passing tab to `truncateReferrer` doesn't seem right. Should we update `isSafeListed` instead to check through all safelists?